auacm/app/modules/problem_manager/views.py

```diff
@@ -23,49 +13,47 @@
 from app.util import serve_response, serve_error, serve_info_pdf, admin_required
 from app.modules.submission_manager.models import Submission
 from app.modules.problem_manager.models import Problem, ProblemData, SampleCase
+from app.modules.competition_manager.models import Competition
 from sqlalchemy.orm import load_only
 from json import loads
 from shutil import rmtree
-
-
-def url_for_problem(problem):
-    """Return the path of the pdf description of a problem"""
-    return os.path.join('problems', str(problem.shortname),
-                        'info.pdf')
-
-
-def is_pid(identifier):
-    try:
-        int(identifier)
-        return True
-    except ValueError:
-        return False
+from time import time
 
 
 @app.route('/problems/<shortname>/info.pdf', methods=['GET'])
 def get_problem_info(shortname):
     """Serve the PDF description of a problem"""
-    pid = database.session.query(Problem).\
-            options(load_only('pid', 'shortname')).\
-            filter(Problem.shortname == shortname).\
-            first().pid
+    pid = (database.session.query(Problem)
+           .options(load_only('pid', 'shortname'))
+           .filter(Problem.shortname == shortname)
+           .first().pid)
     return serve_info_pdf(str(pid))
 
 
 @app.route('/api/problems/<identifier>', methods=['GET'])
 def get_problem(identifier):
-    """Returns the JSON representation of a specific problem"""
+    """
+    Returns the JSON representation of a specific problem
+
+    If the problem is meant to be released with a competition that has not
+    started yet, a 404 error is returned.
+    """
     problem = database.session.query(Problem, ProblemData).join(ProblemData)
+
     if is_pid(identifier):
         problem = problem.filter(Problem.pid == identifier).first()
     else:
-        problem = problem.\
-                  filter(Problem.shortname == identifier).first()
+        problem = problem.filter(Problem.shortname == identifier).first()
+
+    # Hide unreleased problems to non-admins
+    if problem is None or (current_user.admin != 1 and comp_not_released(
+            problem.Problem.comp_release)
```

Comment:
It's not running due to a missed parenthesis here. I'm just going to push that change.